client/app/services/query.js

```diff
@@ -1,10 +1,25 @@
 import moment from 'moment';
 import debug from 'debug';
 import Mustache from 'mustache';
-import { each, zipObject, isEmpty, map, filter, includes, union, uniq, has } from 'lodash';
+import {
+  each, zipObject, isEmpty, map, filter, includes, union, uniq, has,
+  isNull, isUndefined,
+} from 'lodash';
 
 const logger = debug('redash:services:query');
 
+const DATETIME_FORMATS = {
+  // eslint-disable-next-line quote-props
+  'date': 'YYYY-MM-DD',
+  'datetime-local': 'YYYY-MM-DD HH:mm',
+  'datetime-with-seconds': 'YYYY-MM-DD HH:mm:ss',
+};
+
+function normalizeNumericValue(value, defaultValue = null) {
+  const result = parseInt(value, 10);
```

Comment:
Number typed parameter can be a float as well. I'm not sure why we used `parseInt` there in the first place, but maybe it was a non issue because we didn't use `ngModel` with input of type number. But worth fixing this now. Is there a reason to keep this parsing?